docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,131 @@
+pipelines and using Kedro in a monorepo setup.
+
+## Where does Kedro look for code to be located
+
+The only technical constraint for arranging code in the project is that `pipeline_registry.py`
+file must be located in `<your_project>/src/<your_project>` directory, which is where
+it is created by default.
+
+This file must have a `register_pipelines()` function that returns a `tp.Dict[str, Pipeline]`
+mapping from pipeline name to corresponding `Pipeline` object.
+
+Other than that, **Kedro does not impose any constraints on where you should keep files with
+`Pipeline`s, `Node`s, or functions wrapped by `node`**.
+
+```{note}
+You actually can make Kedro look for pipeline registry in a different place by modifying the
+`__main__.py` file of your project, but such advanced customizations are not in scope of this article.
+```
+
+This being the only constraint means that you can, for example:
+* Add `utils.py` file to a pipeline folder and import utilities defined by multiple
+  functions in `nodes.py`.
+* Delete or rename a default `nodes.py` file, split it into multiple files or modules.
+* Instead of having a single `pipeline.py` in your pipeline folder, split it
+  into `historical_pipeline.py` and `inference_pipeline.py`.
+* Instead of registering many pipelines in `register_pipelines()` function one by one,
+  create a few `tp.Dict[str, Pipeline]` objects in different places of the project
+  and then make `register_pipelines()` return a union of those.
+* Store code that has nothing to do with Kedro `Pipeline` and `Node` concepts, or should
+  be reused by multiple pipelines of your project, in a module at the same level as the
+  `pipelines` folder of your project. This scenario is covered in more detail below.
+
+## Common codebase extension scenarios
+
+This section provides examples of how you can handle some common cases of adding more
+code to or around your Kedro project.
+**Provided implementations are by no means the only ways to achieve target scenarios**,
+and serve only illustrative purposes.
+
+### Sharing utilities between pipelines
+
+Oftentimes you have utilities that have to be imported by multiple `pipelines`.
+To keep them as part of a Kedro project, **create a module (for example, `utils`) at the same
+level as the `pipelines` folder**, and organise the functionalities there:
+
+```text
+├── conf
+├── data
+├── notebooks
+└── src
+    ├── my_project
+    │   ├── __init__.py
+    │   ├── __main__.py
+    │   ├── pipeline_registry.py
+    │   ├── settings.py
+    │   ├── pipelines
+    │   └── utils                       <-- Create a module to store your utilities
+    │       ├── __init__.py             <-- Required to import from it
+    │       ├── pandas_utils.py         <-- Put a file with utility functions here
+    │       ├── dictionary_utils.py     <-- Or a few files
+    │       ├── visualization_utils     <-- Or sub-modules to organize even more utilities
```

Comment:
Nit: we use UK spelling over US (https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide)

```suggestion
    │       ├── visualisation_utils     <-- Or sub-modules to organise even more utilities
```

Reply:
Yeah thanks for this GitHub suggestion and all similar above. I saw this styleguide, and even did things like CTRL+F "ize" in a file, but for some reason did not clean it in all at once. So, my bad, fixing now :)